### PR TITLE
feat(feature-flags): wire in experiments to the feature-flag api

### LINKED ISSUE #6890: Render feature-flagging config in the DOM
Depends on mozilla/fxa-shared#41.

[Feature doc](https://docs.google.com/document/d/1pRQuK7GWM3zEZLObZCByDxU_cImduICHrL6SaPSoUkU).

### PR DESCRIPTION
**Pull request by philbooth**
Fixes #6890. Depends on jacobrask/eslint-plugin-sorting#26 to fix the lint failure.

Marking as `WIP` until we've set it up with a working implementation, but this hooks the existing experiment infrastructure into the skeleton feature-flagging API from mozilla/fxa-shared#45.

### CODE REVIEW

File: server/lib/configuration.js [hunk 1]
```diff
@@ -147,6 +147,20 @@ const conf = module.exports = convict({
       'development'
     ]
   },
+  featureFlags: {
+    implementation: {
+      default: 'launch-darkly',
+      doc: 'The feature-flagging implementation to use, e.g. "launch-darkly" or "unleash"',
+      env: 'FEATURE_FLAG_IMPLEMENTATION',
+      format: String
```
Comment (vladikoff) [inline server/lib/configuration.js]: would it make sense to make this a `format` like https://github.com/mozilla/fxa-content-server/blob/b8e01a4ae15a82a310f3a7f4a2a64e27d321cf4b/server/lib/configuration.js#L395 and pick from a list of available options? Also would it make sense to introduce a `none` to turn all flags off, is there a scenario where we would need this?

**Comment by philbooth** [comment]
Note that I've now removed the demo feature-flags from this branch, to start getting it ready for code review.

If anyone's still interested in playing with those, they're in `pb/6890-feature-flags-demo`.

**Comment by philbooth** [comment]
I'm pulling the `WIP` off this and calling it ready for review, although there is also a failing test in CI that I hope someone can help me out with.

But anyway, the `fxa-shared` work landed in mozilla/fxa-shared#61 and this just hooks it in with our existing experiments. Missing feature flags just fall back to the existing experiment config. If you're interested in the demo experiments I was playing around with before, they're still available in the `pb/6890-feature-flags-demo` branch.

@mozilla/fxa-devs r?

And this is the error in CI, which I'll investigate more tomorrow:

```
‣ Created remote session firefox on linux 4.4.0-141-generic (d5aa6c9f-1663-4c81-9d7d-a11036aa54c1)
Error
Error
Error
Screenshot saved at: https://screencap.co.uk/images/8a58103144c2b241b6c2f1b0738da63ccd0f1ea0.png
× firefox on linux 4.4.0-141-generic - pairing - it can pair (12.734s)
    Error: QRTimeout: Error
      at getQrData.then.catch  <tests/functional/pairing.js:83:37>
      at pollForScreenshot  <tests/functional/pairing.js:73:8>
      at Timeout._onTimeout  <tests/functional/pairing.js:87:19>
      at ontimeout  <timers.js:498:11>
      at tryOnTimeout  <timers.js:323:5>
      at Timer.listOnTimeout  <timers.js:290:5>
      at pollForScreenshot  <tests/functional/pairing.js:73:8>
      at Timeout._onTimeout  <tests/functional/pairing.js:87:19>
      at ontimeout  <timers.js:498:11>
      at tryOnTimeout  <timers.js:323:5>
      at Timer.listOnTimeout  <timers.js:290:5>
      at pollForScreenshot  <tests/functional/pairing.js:73:8>
      at Command.<anonymous>  <tests/functional/pairing.js:95:10>
      at Command.<anonymous>  <tests/functional/lib/helpers.js:60:23>
      at Command.<anonymous>  <tests/functional/lib/helpers.js:73:18>
      at Command.<anonymous>  <tests/functional/lib/helpers.js:68:10>
      at Test.it can pair [as test]  <tests/functional/pairing.js:116:10>
      at <src/lib/Test.ts:263:51>
      at Test.it can pair [as test]  <tests/functional/pairing.js:120:15>
      at <src/lib/Test.ts:263:51>

~ firefox on linux 4.4.0-141-generic - pairing - handles invalid clients (bailed)
TOTAL: tested 1 platforms, 0 passed, 1 failed, 1 skipped
```

**Comment by philbooth** [comment]
> ...although there is also a failing test in CI that I hope someone can help me out with.

Evidently that was some kind of yesterday-weirdness, because it's working this morning.

File: app/scripts/lib/experiments/grouping-rules/is-sampled-user.js [hunk 2]
```diff
@@ -27,7 +27,11 @@ module.exports = class IsSampledUserGroupingRule extends BaseGroupingRule {
    * @param {String} env
```
Comment (shane-tomlinson) [inline app/scripts/lib/experiments/grouping-rules/is-sampled-user.js]: This comment is out of date and now takes an object.

**Pull request merged**